connectors/src/api/webhooks/webhook_slack.ts

```diff
@@ -45,55 +46,110 @@ const _webhookSlackAPIHandler = async (
     return res.status(200).send({
       challenge: req.body.challenge,
     });
-  } else if (req.body.type === "event_callback") {
+  }
+
+  if (req.body.type === "event_callback") {
     if (!req.body.team_id) {
-      return res.status(400).send({
-        error: {
+      return apiError(req, res, {
+        api_error: {
+          type: "invalid_request_error",
           message: "Missing team_id in request body",
         },
+        status_code: 400,
       });
     }
-    const slackConfiguration = await SlackConfiguration.findOne({
+    const slackConfigurations = await SlackConfiguration.findAll({
       where: {
         slackTeamId: req.body.team_id,
       },
     });
-    if (!slackConfiguration) {
-      return res.status(404).send({
-        error: {
+    if (slackConfigurations.length === 0) {
+      return apiError(req, res, {
+        api_error: {
+          type: "connector_configuration_not_found",
           message: `Slack configuration not found for teamId ${req.body.team_id}`,
         },
+        status_code: 404,
```

Comment:
I am actually not sure we want to 404 here. If we don't have it and Slack tells us about it, should we just tell Slack to stop pinging us? Up to you.